modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/processor/AbstractSQLProcessor.java

```diff
@@ -330,11 +331,17 @@ public ODataResponse createEntity(final PostUriInfo uriInfo, final InputStream c
             updateCreateEventHandlerContext(handlerContext, connection, uriInfo, entry);
             this.odata2EventHandler.beforeCreateEntity(uriInfo,
                     requestContentType, contentType, entry, handlerContext);
+            if(handlerContext.containsKey(ERROR_DOCUMENT) && handlerContext.get(ERROR_DOCUMENT) != null) {

```

Comment:
Could you move the conditions to a separate method called isErrorDocumentPresent() and then use the method if(isErrorDocumentPresent()) -> return get(ERROR_DOCUMENT) for all checks below?